ЗАДАЧА 1:
Даны две независимые выборки. Не соблюдается условие нормальности
x1 380,420, 290
y1 140,360,200,900
Сделайте вывод по результатам, полученным с помощью функции
H0: m = m0
H1: m != m0
a = 0.05

In [1]:
import numpy as np
import scipy.stats as stats

In [6]:
x1 = np.array([380, 420, 290])
x2 = np.array([140, 360, 200, 900])

In [7]:
stats.mannwhitneyu(x1, x2)

MannwhitneyuResult(statistic=8.0, pvalue=0.6285714285714286)

In [ ]:
pvalue (0.629) > a(0.05)
Вывод: Гипотеза H0 (нет статистических значимых различий) верна при а = 0.05

ЗАДАЧА 2:
Исследовалось влияние препарата на уровень давления пациентов. Сначала измерялось давление до приема 
препарата, потом через 10 минут и через 30 минут. Есть ли статистически значимые различия?
1е измерение до приема препарата: 150, 160, 165, 145, 155
2е измерение через 10 минут: 140, 155, 150, 130, 135
3е измерение через 30 минут: 130, 130, 120, 130, 125
H0: m = m0
H1: m != m0
a = 0.05    

In [8]:
izm1 = np.array([150, 160, 165, 145, 155])
izm2 = np.array([140, 155, 150, 130, 135])
izm3 = np.array([130, 130, 120, 130, 125])

In [9]:
stats.friedmanchisquare(izm1, izm2, izm3)

FriedmanchisquareResult(statistic=9.578947368421062, pvalue=0.00831683351100441)

pvalue(0.00831683351100441) < a(0.05)
Вывод: Верна гипотеза Н1 (есть статистических значимые различия) при а = 0ю05

ЗАДАЧА 3:
Сравните 1 и 2 е измерения, предполагая, что 3го измерения через 30 минут не было.
H0: m = m0
H1: m != m0
a = 0.05

In [10]:
izm1 = np.array([150, 160, 165, 145, 155])
izm2 = np.array([140, 155, 150, 130, 135])

In [11]:
stats.wilcoxon(izm1, izm2)

WilcoxonResult(statistic=0.0, pvalue=0.0625)

pvalue (0.0625) > a(0.05)
Вывод: Гипотеза H0 (нет статистических значимых различий) верна при а = 0.05

ЗАДАЧА 4:
Даны 3 группы учеников плавания.
В 1 группе время на дистанцию 50 м составляют:
56, 60, 62, 55, 71, 67, 59, 58, 64, 67
Вторая группа : 57, 58, 69, 48, 72, 70, 68, 71, 50, 53
Третья группа: 57, 67, 49, 48, 47, 55, 66, 51, 54 
H0: m = m0
H1: m != m0
a = 0.05

In [12]:
gr1 = np.array([56, 60, 62, 55, 71, 67, 59, 58, 64, 67])
gr2 = np.array([57, 58, 69, 48, 72, 70, 68, 71, 50, 53])
gr3 = np.array([57, 67, 49, 48, 47, 55, 66, 51, 54])

In [13]:
stats.kruskal(gr1, gr2, gr3)

KruskalResult(statistic=5.465564058257224, pvalue=0.0650380998590494)

pvalue (0.0650380998590494) > a(0.05)
Вывод: Гипотеза H0 (нет статистических значимых различий) верна при а = 0.05

ЗАДАЧА 5:
Заявляется, что партия изготавливается со средним арифметическим 2,5 см. 
Проверить данную гипотезу, если известно, что размеры изделий подчинены нормальному закону 
распределения. Объем выборки 10, уровень статистической значимости 5%
2.51, 2.35, 2.74, 2.56, 2.40, 2.36, 2.65, 2.7, 2.67, 2.34 
H0: m = m0
H1: m != m0
a = 0.05
n = 10
m0 = 2.5
t0 = 2.262 (табличное значение)

In [14]:
x = np.array([2.51, 2.35, 2.74, 2.56, 2.40, 2.36, 2.65, 2.7, 2.67, 2.34])

1.среднее арифметическое

In [28]:
arr = [2.51, 2.35, 2.74, 2.56, 2.40, 2.36, 2.65, 2.7, 2.67, 2.34]
m1 = sum(arr)/len(arr)
m1

2.5279999999999996

In [29]:
np.mean(x)

2.5279999999999996

2.среднее квадратичное отклонение

In [40]:
disp = (sum((x - m1)**2 for x in arr)/(10 - 1))**0.5
disp

0.1572542173961923

In [34]:
np.std(x, ddof = 1)

0.1572542173961923

In [41]:
m0 = 2.5
t = (m1 - m0)/(disp/(10**0.5))
t

0.5630613661802959

t(0.5935187931919458) < t0(2.262)
Вывод: Принимаем гипотезу Н0

In [24]:
stats.ttest_1samp(x, 2.5)

Ttest_1sampResult(statistic=0.5630613661802959, pvalue=0.5871439993940628)

In [ ]:
pvalue(0.5871439993940628) > a(0.05)
Вывод: Принимаем гипотезу Н0